client/db/src/cache/list_cache.rs

```diff
@@ -156,8 +158,8 @@ impl<Block: BlockT, T: CacheItemT, S: Storage<Block, T>> ListCache<Block, T, S>
 
 			// BUT since we're not guaranteeing to provide correct values for forks
 			// behind the finalized block, check if the block is finalized first
-			if !chain::is_finalized_block(&self.storage, at, Bounded::max_value())? {
-				return Ok(None);
+			if !chain::is_finalized_block(&self.storage, &at, Bounded::max_value())? {
+				return Err(ClientError::NotInFinalizedChain);
```

Comment:
OK - will try to sync. But there were no cache on full clients before this PR. On light clients finalized entries were pruned => we have never touch any entries for finalized blocks. So the issue could only be introduced by this PR itself.

The only case when this could theoretically happen is when we're trying to import non-canonical block with number <= best finalized. I've asked about this case in Riot && got confirmation that we never should expect to import such blocks.

Reply:
This assumption seems fair.